### PR TITLE
CDAP-5131 bugfix for filtering plugin class by type in artifactStore

### PR DESCRIPTION
**Pull request by gsps1**
JIRA : https://issues.cask.co/browse/CDAP-5131

### CODE REVIEW

**Comment by gsps1** [comment]
Build running : http://builds.cask.co/browse/CDAP-DUT3920-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java [hunk 1]
```diff
@@ -444,7 +448,14 @@ public ArtifactData apply(DatasetContext<Table> context) throws Exception {
         @Override
         public SortedMap<ArtifactDescriptor, List<PluginClass>> apply(DatasetContext<Table> context) throws Exception {
           Table table = context.get();
-          SortedMap<ArtifactDescriptor, List<PluginClass>> result = getPluginsInArtifact(table, parentArtifactId);
+          SortedMap<ArtifactDescriptor, List<PluginClass>> result =
+            getPluginsInArtifact(table, parentArtifactId, new Predicate<PluginClass>() {
+              @Override
+              public boolean apply(@Nullable PluginClass input) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java]: Is `input` `Nullable`?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java [hunk 2]
```diff
@@ -837,12 +853,19 @@ private void deleteMeta(Table table, Id.Artifact artifactId, byte[] oldData) thr
     // include any plugin classes that are inside the artifact itself
     ArtifactData parentData = gson.fromJson(Bytes.toString(parentDataBytes), ArtifactData.class);
     Set<PluginClass> parentPlugins = parentData.meta.getClasses().getPlugins();
-    if (!parentPlugins.isEmpty()) {
+
+    Set<PluginClass> filteredPlugins = new HashSet<>();
+    for (PluginClass pluginClass : parentPlugins) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java]: You can do it in a more fluent way:

```java
Set<PluginClass> filteredPlugins = Sets.newLinkedHashSet(Iterables.filter(parentPlugins, filter))
```

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java [hunk 3]
```diff
@@ -837,12 +853,19 @@ private void deleteMeta(Table table, Id.Artifact artifactId, byte[] oldData) thr
...
+      if (filter.apply(pluginClass)) {
+        filteredPlugins.add(pluginClass);
+      }
+    }
+
+    if (!filteredPlugins.isEmpty()) {
       Location parentLocation = locationFactory.create(parentData.locationURI);
       ArtifactDescriptor descriptor = new ArtifactDescriptor(artifactId.toArtifactId(), parentLocation);
-      result.put(descriptor, Lists.newArrayList(parentPlugins));
+      result.put(descriptor, Lists.newArrayList(filteredPlugins));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java]: Why need to copy to a list? Can't we just use the `filteredPlugins` set?

Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java]: was done before your PR, but a Set probably makes more sense

**Comment by albertshau** [comment]
minor comments, otherwise lgtm

**Comment by gsps1** [comment]
addressed comments, build green - https://builds.cask.co/browse/CDAP-DUT3920-2

**Comment by albertshau** [comment]
lgtm

**Pull request merged**